test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -273,17 +282,19 @@ public void MatrixFactorizationInMemoryDataZeroBaseIndex()
             // Create a matrix factorization trainer which may consume "Value" as the training label, "MatrixColumnIndex" as the
             // matrix's column index, and "MatrixRowIndex" as the matrix's row index.
             var mlContext = new MLContext(seed: 1, conc: 1);
-            var pipeline = mlContext.Recommendation().Trainers.MatrixFactorization(
-                nameof(MatrixElementZeroBased.MatrixColumnIndex),
-                nameof(MatrixElementZeroBased.MatrixRowIndex),
-                nameof(MatrixElementZeroBased.Value),
-                advancedSettings: s =>
-                {
-                    s.NumIterations = 100;
-                    s.NumThreads = 1; // To eliminate randomness, # of threads must be 1.
-                    s.K = 32;
-                    s.Eta = 0.5;
-                });
+
+            var options = new MatrixFactorizationTrainer.Options
+            {
+                MatrixColumnIndexColumnName = nameof(MatrixElement.MatrixColumnIndex),
+                MatrixRowIndexColumnName = nameof(MatrixElement.MatrixRowIndex),
+                LabelColumnName = nameof(MatrixElement.Value),
+                NumIterations = 100,
+                NumThreads = 1,
```

Comment:
May we have my old comment back? :)

Reply:
fixed

---
In reply to: [250022806](https://github.com/dotnet/machinelearning/pull/2204#discussion_r250022806) [](ancestors = 250022806)